app/com/arpnetworking/metrics/portal/alerts/impl/CassandraAlertRepository.java

```diff
@@ -0,0 +1,245 @@
+import models.internal.impl.DefaultQuantity;
+import models.internal.impl.DefaultQueryResult;
+import org.joda.time.Period;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+
+/**
+ * Implementation of an {@link AlertRepository} that stores the data in Cassandra.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class CassandraAlertRepository implements AlertRepository {
+    /**
+     * Public constructor.
+     *
+     * @param cassandraSession a Session to use to query data
+     * @param mappingManager a MappingManager providing ORM for the Cassandra objects
+     */
+    @Inject
+    public CassandraAlertRepository(final Session cassandraSession, final MappingManager mappingManager) {
+        _cassandraSession = cassandraSession;
+        _mappingManager = mappingManager;
+    }
+
+    @Override
+    public void open() {
+        assertIsOpen(false);
+        LOGGER.debug().setMessage("Opening alert repository").log();
+        _isOpen.set(true);
+    }
+
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing alert repository").log();
+        _isOpen.set(false);
+        _cassandraSession.close();
+    }
+
+    @Override
+    public Optional<Alert> get(final UUID identifier, final Organization organization) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Getting alert")
+                .addData("alertId", identifier)
+                .addData("organization", organization)
+                .log();
+        final Mapper<models.cassandra.Alert> mapper = _mappingManager.mapper(models.cassandra.Alert.class);
+        final models.cassandra.Alert cassandraAlert = mapper.get(identifier);
+
+        if (cassandraAlert == null) {
+            return Optional.empty();
+        }
+
+        return Optional.of(convertFromCassandraAlert(cassandraAlert));
+    }
+
+    @Override
+    public AlertQuery createQuery(final Organization organization) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Preparing query")
+                .addData("organization", organization)
+                .log();
+        return new DefaultAlertQuery(this, organization);
+    }
+
+    @Override
+    public QueryResult<Alert> query(final AlertQuery query) {
+        final Mapper<models.cassandra.Alert> mapper = _mappingManager.mapper(models.cassandra.Alert.class);
+        final models.cassandra.Alert.AlertQueries accessor = mapper.getManager().createAccessor(models.cassandra.Alert.AlertQueries.class);
+        final Result<models.cassandra.Alert> result = accessor.getAlertsForOrganization(query.getOrganization().getId());
+        final List<models.cassandra.Alert> allAlerts = result.all();
```

Comment:
Should we either:

1) have a way to have Cassandra perform the filtering.
2) have Cassandra return a stream/cursor.

Whatever you think is better I'm fine with a `TODO` here and an issue.